test/run_test.sh

```diff
@@ -64,6 +64,15 @@ $PYCMD test_cuda.py $@
 echo "Running NCCL tests"
 $PYCMD test_nccl.py $@
 
+echo "Running C++ Extensions tests"
+cd cpp_extensions
+$PYCMD setup.py install --record installed_files.txt
```

Comment:
Maybe we could test if `virtualenv` is available and skip the tests otherwise? This would be a very safe way install the modules. Alternatively, I think we might try playing with arguments that can be given to `install` e.g. `--install-base` or `--home`, to install the package locally, and then add it to `sys.path` or `PYTHONHOME` or something like that. This also makes the cleanup much easier since we can just `rm -rf install_dir`

Reply:
The problem with virtualenv is that you'll lose access to the PyTorch you already installed. Maybe there is a way to make stacked Python environments; that would definitely the preferred resolution.

Reply:
Good point. I think a solution that would let us see both is to install in a local directory and add it to `sys.path` somehow (there's probably a command-line argument that sets it)

Reply:
`pip install -t target_dir`

Reply:
Do we really want pip here? I don't see any deps that need installing, and it would be a disaster if pip concluded that it needed to install PyTorch.